File: jclasslib/src/org/gjt/jclasslib/browser/BrowserAboutDialog.java
/*
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public
    License as published by the Free Software Foundation; either
    version 2 of the license, or (at your option) any later version.
*/

package org.gjt.jclasslib.browser;

import org.gjt.jclasslib.util.GUIHelper;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

/**
 * About dialog.
 *
 * @author <a href="mailto:jclasslib@ej-technologies.com">Ingo Kegel</a>
 * @version $Revision: 1.2 $ $Date: 2005-01-14 15:01:03 $
 */
public class BrowserAboutDialog extends JDialog {

    private JButton btnOk;

    /**
     * Constructor.
     *
     * @param parent parent frame.
     */
    public BrowserAboutDialog(JFrame parent) {
        super(parent);
        setupControls();
        setupComponent();
    }

    private void setupComponent() {

        setModal(true);
        setTitle("About the jclasslib bytecode viewer");
        setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
        setSize(400, 250);
        GUIHelper.centerOnParentWindow(this, getOwner());

        JComponent contentPane = (JComponent)getContentPane();
        contentPane.setLayout(new GridBagLayout());
        GridBagConstraints gc = new GridBagConstraints();
        gc.insets = new Insets(10, 5, 0, 5);
        gc.gridx = 0;
        gc.gridy = GridBagConstraints.RELATIVE;

        gc.anchor = GridBagConstraints.CENTER;
        gc.weightx = 1;

        JLabel label = new JLabel("jclasslib bytecode viewer");
        label.setFont(label.getFont().deriveFont(Font.BOLD));
        contentPane.add(label, gc);
        gc.insets.top = 5;
        contentPane.add(new JLabel("Version " + BrowserApplication.APPLICATION_VERSION), gc);
        contentPane.add(new JLabel("Copyright ej-technologies GmbH, 2001-2005"), gc);
        contentPane.add(new JLabel("Licensed under the General Public License"), gc);

        gc.weighty = 0;
        gc.insets.top = 20;
        gc.insets.bottom = 5;
        gc.fill = GridBagConstraints.NONE;
        contentPane.add(btnOk, gc);

        Dimension size = contentPane.getPreferredSize();
        size.width += 100;
        contentPane.setPreferredSize(size);
        pack();
        setResizable(false);

    }

    private void setupControls() {

        btnOk = new JButton("Ok");
        btnOk.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent event) {
                setVisible(false);
                dispose();
            }
        });
    }
}

File: modules/browser/src/main/java/org/gjt/jclasslib/browser/BrowserAboutDialog.kt
/*
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public
    License as published by the Free Software Foundation; either
    version 2 of the license, or (at your option) any later version.
*/

package org.gjt.jclasslib.browser

import com.install4j.api.launcher.Variables
import org.gjt.jclasslib.util.DefaultAction
import org.gjt.jclasslib.util.GUIHelper
import java.awt.Font
import java.awt.GridBagConstraints
import java.awt.GridBagLayout
import java.awt.Insets
import java.io.IOException
import java.util.*
import javax.swing.JComponent
import javax.swing.JDialog
import javax.swing.JFrame
import javax.swing.JLabel

class BrowserAboutDialog(parent: JFrame) : JDialog(parent) {

    private val okAction = DefaultAction("OK") {
        isVisible = false
        dispose()
    }

    init {
        setupComponent()
    }

    private fun setupComponent() {
        (contentPane as JComponent).apply {
            border = GUIHelper.WINDOW_BORDER
            layout = GridBagLayout()

            val gc = GridBagConstraints().apply {
                insets = Insets(10, 50, 5, 50)
                gridx = 0
                gridy = GridBagConstraints.RELATIVE
                anchor = GridBagConstraints.CENTER
                weightx = 1.0
            }

            add(JLabel("jclasslib bytecode viewer").apply {
                font = font.deriveFont(Font.BOLD)
            }, gc)
            gc.insets.top = 0
            add(JLabel("Version " + getVersion()), gc)
            add(JLabel("Copyright ej-technologies GmbH, 2001-" + Calendar.getInstance().get(Calendar.YEAR)), gc)
            add(JLabel("Licensed under the General Public License"), gc)

            contentPane.add(okAction.createTextButton().apply {
                this@BrowserAboutDialog.getRootPane().defaultButton = this
            }, GridBagConstraints().apply {
                insets = Insets(20, 0, 0, 0)
                gridx = 0
                gridy = GridBagConstraints.RELATIVE
                anchor = GridBagConstraints.CENTER
                gc.fill = GridBagConstraints.NONE
            })
        }

        pack()
        isModal = true
        isResizable = false
        title = "About the jclasslib bytecode viewer"
        GUIHelper.centerOnParentWindow(this, owner)
        defaultCloseOperation = JDialog.DISPOSE_ON_CLOSE
    }

    private fun getVersion(): String {
        try {
            return Variables.getCompilerVariable("sys.version")
        } catch (e: IOException) {
            return "<unknown>"
        }
    }
}
